lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -170,33 +170,47 @@ public static function get_theme_data() {
 	 *
 	 * It can also create and return a new draft CPT.
 	 *
-	 * @param bool  $should_create_cpt Whether a new CPT should be created if no one was found.
-	 *                                 False by default.
-	 * @param array $post_status_filter Filter CPT by post status.
-	 *                                  ['publish'] by default, so it only fetches published posts.
+	 * @param WP_Theme $theme             The theme object.
+	 *                                    If empty, it defaults to the current theme.
+	 * @param bool     $should_create_cpt Whether a new CPT should be created if no one was found.
+	 *                                    False by default.
+	 * @param array    $post_status_filter Filter CPT by post status.
+	 *                                    ['publish'] by default, so it only fetches published posts.
 	 *
 	 * @return array Custom Post Type for the user's origin config.
 	 */
-	private static function get_user_data_from_custom_post_type( $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
+	public static function get_user_data_from_custom_post_type( $theme, $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
+		if ( ! $theme instanceof WP_Theme ) {
+			$theme = wp_get_theme();
+		}
 		$user_cpt         = array();
 		$post_type_filter = 'wp_global_styles';
-		$recent_posts     = wp_get_recent_posts(
-			array(
-				'numberposts' => 1,
-				'orderby'     => 'date',
-				'order'       => 'desc',
-				'post_type'   => $post_type_filter,
-				'post_status' => $post_status_filter,
-				'tax_query'   => array(
-					array(
-						'taxonomy' => 'wp_theme',
-						'field'    => 'name',
-						'terms'    => wp_get_theme()->get_stylesheet(),
-					),
+		$args             = array(
+			'numberposts' => 1,
+			'orderby'     => 'date',
+			'order'       => 'desc',
+			'post_type'   => $post_type_filter,
+			'post_status' => $post_status_filter,
+			'tax_query'   => array(
+				array(
+					'taxonomy' => 'wp_theme',
+					'field'    => 'name',
+					'terms'    => $theme->get_stylesheet(),
 				),
-			)
+			),
 		);
 
+		$cache_key = sprintf( 'wp_global_styles_post_type_%s', md5( serialize( $args ) ) );
+		$post_id   = wp_cache_get( $cache_key );
+		if ( false !== $post_id ) {
+			if ( (int) $post_id > 0 ) {
+				return get_post( $post_id, ARRAY_A );
+			} elseif ( -1 === $post_id && ! $should_create_cpt ) {
```

Comment:
What do you think about changing this into multiple if statements?

```
if ( (int) $post_id > 0 ) {
     return get_post( $post_id, ARRAY_A );
}

// Special case: '-1' is a results not found.
if ( -1 === $post_id && ! $should_create_cpt ) {
     return $user_cpt;
}
```

Reply:
Changed it in 934143c09d